### PR TITLE
Reenable block bindings

### PR DESCRIPTION
**Pull request by StefanIliev545**
### Why this change is needed

The block binding is disabled on main which is insecure. 

### What changes were made as part of this PR

Please provide a high level list of the changes made

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/ten-protocol/ten-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: contracts/src/management/ManagementContract.sol [hunk 1]
```diff
@@ -114,20 +114,20 @@ contract ManagementContract is Initializable, OwnableUpgradeable {
         return isBundleSaved[bundleHash];
     }
 
-    function addCrossChainMessagesRoot(bytes32 _lastBatchHash, bytes32 blockHash, uint256 blockNum, bytes[] memory crossChainHashes, bytes calldata signature, uint256 rollupNumber, bytes32 forkID) external {
-        /*  if (block.number > blockNum + 255) {
-              revert("Block binding too old");
-          }
+    function blockBinding(uint256 blockNum) public view returns(bytes32) {
```
Comment (tudor-malene) [inline contracts/src/management/ManagementContract.sol:117]: is this used?

**Review by tudor-malene — COMMENTED** [review]
LGTM ( I think).
Small comments

File: go/host/l1/publisher.go [hunk 2]
```diff
@@ -308,14 +310,42 @@ func (p *Publisher) PublishCrossChainBundle(bundle *common.ExtCrossChainBundle,
...
 	tx, err := managementCtr.AddCrossChainMessagesRoot(transactor, [32]byte(bundle.LastBatchHash.Bytes()), bundle.L1BlockHash, bundle.L1BlockNum, bundle.CrossChainRootHashes, bundle.Signature, rollupNum, forkID)
 	if err != nil {
-		if !errors.Is(err, errutil.ErrCrossChainBundleRepublished) {
+		if errors.Is(err, errutil.ErrCrossChainBundleRepublished) {
 			p.logger.Info("Cross chain bundle already published. Proceeding without publishing", log.ErrKey, err, log.BundleHashKey, bundle.LastBatchHash)
 			return nil
 		}
+
+		rpcErr, ok := err.(rpc.Error)
+		if ok && strings.Contains(rpcErr.Error(), errutil.ErrBlockBindingMismatch.Error()) {
```
[the remark this answers is not in the document]
Comment (StefanIliev545) [reply go/host/l1/publisher.go]: Nope, the errors returned from geth are with some different wrapping so this fails; Regardless this is debug code that i removed

File: contracts/src/management/ManagementContract.sol (hunk 1 above)
Comment (StefanIliev545) [reply contracts/src/management/ManagementContract.sol:117]: Was used for debugging and figuring out if im crazy, will remove

**Review by tudor-malene — APPROVED** [review]
lgtm

**Pull request merged**